### PR TITLE
[SAI-Qualify]scan Swss log and upload to kusto

### PR DESCRIPTION
**Pull request by ms-junyi**
### Description of PR

1. collect sairedis logs from different types of machines
2. analyze the logs, extract the key and value organized by json
3. ingest json file to the kusto table SwssInvocationReport

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

### CODE REVIEW

File: test_reporting/sai_swss_invocations.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+from report_data_storage import KustoConnector
+
+'''
+for more detail, you can see
+https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/373699/sairedis-log-scanner
+'''
+
+
+ngsdevice_type = "ToRRouter"
+sai_path = "/data/sonic-mgmt-int/SAI/inc/"
```
Comment (richardyu-ms) [inline test_reporting/sai_swss_invocations.py:21]: This folder is self defined or cloned automatically?
 Need to check the existence and report error message with details like where is the repo and command, or clone it in your code

File: test_reporting/sai_swss_invocations.py [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+json_log_path = "/data/sonic-mgmt-int/test_reporting/test"
```
Comment (richardyu-ms) [inline test_reporting/sai_swss_invocations.py:22]: Same as above, and need more comments in the code

File: test_reporting/sai_swss_invocations.py [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+sai_obj_feature_map = {}
+swss_device_log_items = {
```
Comment (richardyu-ms) [inline test_reporting/sai_swss_invocations.py:24]: Put those data into a yaml and upload and use it as a sample?
Same as others, need more comments

File: test_reporting/sai_swss_invocations.py [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    },
+    "/data/sonic-mgmt-int/test_reporting/tmp/PO_ams21-0101-0602-20t0": {
+        'os_version': '20191130.85',
+        'deployment_type':  "Pilotfish",
+        'deployment_subtype': 'Others',
+        'device': 'ams21-0101-0602-20t0',
+    },
+}
+
+sample_log = '2021-11-12.17:41:22.444222|r|SAI_OBJECT_TYPE_ROUTE_ENTRY:{"dest":"2603:10b6:620::/45","switch_id":"oid:0x21000000000000","vr":"oid:0x3000000000042"}'
```
Comment (richardyu-ms) [inline test_reporting/sai_swss_invocations.py:111]: This should be a comment

File: test_reporting/sai_swss_invocations.py [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+operation_map = {'r': 'remove', 'c': 'create',
+                 'g': 'get', 's': 'set', 'q': 'query'}
+# skip G, which is used to get the return value
+operation_features = ['r', 'c', 'g', 's']
```
Comment (richardyu-ms) [inline test_reporting/sai_swss_invocations.py:115]: Comments for each operation

File: test_reporting/sai_swss_invocations.py [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+    onlyfiles = []
+    for f in listdir(path):
+        if isfile(join(path, f)) and name_pattern in f and exclusive_pattern not in f:
+            onlyfiles.append(join(path, f))
+    # for i in onlyfiles:
+    #     print(i)
+    return onlyfiles
+
+
+def generate_sai_feature_file_map_from_header_files(files):
```
Comment (richardyu-ms) [inline test_reporting/sai_swss_invocations.py:133]: Need a sample /comments for what will be the feature

**Comment by richardyu-ms** [comment]
Please add more comments for method and module variables.
Please add more log message on critical steps

File: test_reporting/sai_swss_invocations.py [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+from os.path import isfile, join
+import shutil
+
+from report_data_storage import KustoConnector
+import yaml
+'''
+for more detail, you can see
+https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/373699/sairedis-log-scanner
+'''
+parser = argparse.ArgumentParser()
```
Comment (richardyu-ms) [inline test_reporting/sai_swss_invocations.py:16]: please add those links for getting the parameters into a method that should be invoked when start running this python script.
You can reference report_uploader.py

**Review by richardyu-ms — DISMISSED** [review]

**Comment by richardyu-ms** [comment]
please resove the conflict

**Review by richardyu-ms — DISMISSED** [review]

**Comment by wangxin** [comment]
@ms-junyi Can you fix the LGTM alerts? And for the new files added by you, can you also fix the pre-commit alarmed issues?

**Pull request closed without merging**